AZ3166/src/cores/arduino/cli/console_cli.cpp

```diff
@@ -245,6 +349,90 @@ static void dps_uds_command(int argc, char **argv)
     }
 }
 
+static void az_iotdps_command(int argc, char **argv)
+{
+    char *dpsIdScope = NULL;
+    char *symmetricKey = NULL;
+    char *deviceId = NULL;
+
+    if (argc == 1 || argv[1] == NULL) 
+    {
+        Serial.printf("Usage: set_az_iotdps <connection string>. Please provide the connection string of the IoT Central.\r\n");
+        return;
+    }
+
+    MAP_HANDLE connection_string_values_map;
+    if ((connection_string_values_map = connectionstringparser_parse_from_char(argv[1])) == NULL)
+    {
+        Serial.printf("Tokenizing failed on connectionString");
+        return;
+    }
+    const char *_dpsIdScope = Map_GetValueFromKey(connection_string_values_map, IOTHUBDPS_IDSCOPE);
+    const char *_symmetricKey = Map_GetValueFromKey(connection_string_values_map, IOTHUBDPS_SYMMETRICKEY);
+    const char *_deviceId = Map_GetValueFromKey(connection_string_values_map, IOTHUBDPS_DEVICEID);
+    if (_dpsIdScope)
+    {
+        mallocAndStrcpy_s(&dpsIdScope, _dpsIdScope);
+    }
+    else
+    {
+        Serial.printf("Couldn't find %s in connection string", IOTHUBDPS_IDSCOPE);
+        return;
+    }
+    if (_symmetricKey)
+    {
+        mallocAndStrcpy_s(&symmetricKey, _symmetricKey);
+    }
+    else
+    {
+        Serial.printf("Couldn't find %s in connection string", IOTHUBDPS_SYMMETRICKEY);
+        return;
+    }
+    if (_deviceId)
+    {
+        mallocAndStrcpy_s(&deviceId, _deviceId);
+    }
+    else
+    {
+        Serial.printf("Couldn't find %s in connection string", IOTHUBDPS_DEVICEID);
+        return;
+    }  
+
+    char *sasKey = NULL;
+    if (GenSasToken(symmetricKey, deviceId, &sasKey))
+    {
+        // Failed
+        Serial.printf("Couldn't generate SAS Key from %s in connection string", IOTHUBDPS_SYMMETRICKEY);
+        return;
+    }
+
+    char *dcs = (char*)calloc(AZ_IOT_HUB_MAX_LEN, 1);
+    if (dcs == NULL)
+    {
+        Serial.printf("INFO: Set Iot Central connection string failed. Please try again.\r\n");
+        return;
+    }
+    snprintf(dcs, AZ_IOT_HUB_MAX_LEN, 
+             "DPSEndpoint=global.azure-devices-provisioning.net;IdScope=%s;DeviceId=%s;SymmetricKey=%s",
+             dpsIdScope,
+             deviceId,
+             sasKey);
+    int len = strlen(dcs) + 1;
+    if (len == 0 || len > AZ_IOT_HUB_MAX_LEN)
+    {
+        Serial.printf("Invalid IoT Central connection string.\r\n");
+        free(dcs);
+        return;
+    }
+    int result = write_eeprom(dcs, AZ_IOT_HUB_ZONE_IDX);
+    if (result == 0)
+    {
+        Serial.printf("INFO: Set Iot Central connection string successfully.\r\n");

```

Comment:
Same here. This command is used for generic DPS config, not for IoT Central only.